### ISSUE TITLE
User Access to Necessary Customization

### ISSUE THREAD

**Issue by WayneEDick** [w3c/low-vision-a11y-tf#74]
<h1>User Centered Access to Presentation</h1>

<p>Now that we are moving from defining user needs and policy gaps to creating success criteria, I think we at the LVTF should take a global look at what we have done. Looking at our list of user needs there is one that rises above all others, and it is not on the list. It is the need for full access to modify the visual interface to satisfy the user’s specific visual limitations. Maybe we should make it our first guideline. </p>

<h2>The Guideline Level Need </h2>

<p>An automated method to present the author's content in a format that meets the exact visual requirements of each user, is the best and only method to meet the needs of all people with low vision. Our research on the diversity of low vision functionality documents this fact.</p>

<p>Full user choice is necessary. It should be available to all people with some capability of using a visual interface. We should state it as our basic guideline to visual success criteria.</p>

<h3>Low Vision Guideline #1 (Customization): </h3>

<p>Users must be capable of adjusting the visual interface to meet their exact visual needs.</p>

<h2>State of the Art</h2>

<p>All visual assistive technologies for low vision approximate the goal of complete customization. </p>

<ol>
<li>Screen magnifiers try to meet visual access by addressing size, smoothing enlarged text, color change and many other aspects of visual interface. </li>
<li>User agents provide methods for supporting global color, font size and font family choices as well as zoom and text-only resize.</li>
<li>Operating systems supply support for the user’s visual preferences.</li>
<li>Alternative style sheets (CSS and XSL) provide some customization for HTML and XML data.</li>
</ol>

<p>To date no assistive technology is available for people with low vision that addresses the customization issue completely. </p>

<h2>The Problem</h2>

<p>Given current practices of content creation and user agent interpretation, there appears to be no automated way to customize user interfaces. Any author using current technology can create a user interface that is perfect for one user, but no author can write an individual user interface for every person in the world. Automation is essential. The following are necessary and sufficient conditions for presentation automation:   </p>

<ol>
<li>Content must be flexible enough to support extreme modification of visual presentation without loss of meaning, and</p>
<li>User agents must provide a way to process user presentation requests. </li>
</ol>

<h2>How to Fix It.</h2>

<p>It is already possible to create HTML content that meets the flexibility goals. HTML that supports responsive design can be restyled to support the size, contrast, brightness, color, spacing, font style and box model customization needed for low vision. This gives us a general model.</p>

<p>We need access to three components: Visual Needs Profiles, Flexible Content and User Agents that can apply the User Needs Profiles to create visual presentation.</p>

<h3>Visual Needs Profiles </h3>

<p>Users must have the capability to express their visual presentation needs and have them converted to a declarative programming language that can be interpreted by users and user agents. This profile format should be independent of the content or user agent in use. A profile might look like a style sheet, but be more generic and readable. It might look like this:</p>

<dl>
<dt>Main Region</dt>
<dd>Foreground Color: off white;</dd>
<dd>Background Color: Charcoal;</dd>
<dd>Font Family: Verdana;</dd>
<dd>Font Size: 0.666 in; (1.65 cm) </dd>
<dd>…</dd>
</dt>
<dt>Call Out Region (like an aside)</dt>
<dd>Foreground Color: Light Grey;</dd>
<dd>Background Color: Black;</dd>
<dd>Font Size: 0.5 in; (1.25 cm)</dd>
<dd>…</dd>
</dt>
</dl>

<p>Assistive technology is needed to help users choose their profiles. At present there is little motivation to create this technology because content and user agents do not support user choice. The LVTF can address this problem with success criteria that require flexible content and user agent interpreters of User Needs Profiles.</p>

<h3>Flexible Content </h3>

<p>Size and layout are the difficult problems to solve for user access to visual presentation. Users need to enlarge text significantly (up to 1000% or more) and / or reduce the format to one column with all text word wrapped.</p>

<p>Mobile technology has already forced a solution to the size problem. Presenting information on small screens with normal sizes is topologically equivalent to presenting information on larger screens using large sizes. Mobile solutions need to be applied to adaptations for low vision. The LVTF can create success criteria that delineate how to do this.</p>

<p>The mobile solution to the size problem has also introduced the solution to the formatting problem. Most mobile web data is presented in readable and navigable single column format. Again the LVTF can create success criteria that address this issue directly.</p>

<p>Other requirements such as color, font family and box model are much easier to solve once the size and format issues are addressed. This should be no problem for content developers, but success criteria that prohibit barriers to flexibility may be required.</p>

<h3>User Agents Interpretation of User Needs Profiles</h3>

<p>User agents already have the capability of handling presentation profiles. They are the best programs in the world for doing this. The LVTF will probably have to state explicitly that user agents must support User Needs Profiles. I am not sure what success criteria are needed for this support, but our previous gap analysis that included UAAG should guide us. </p>

<h2>Conclusion</h2>

<p>I strongly urge the LVTF to adopt (the capability to adjust the visual interface to meet personal needs) as the primary goal for visual access. All of our other user needs identify specific instances of access to presentation. The whole is necessary, but I think it is vital to identify full user access to presentation as the foundational need. </p>

<p>I have thought about this as a user and scientist and I think the model I present can serve as the basic framework for addressing the entire problem. I will be happy to accept comments and correction.</p>

<p></p>

**Comment by awkawk** [comment]
Wayne,
Very interesting.  I do think that many of the the individual items you mention in here are covered in the user needs (although we should make sure that everything, like font smoothing, etc.) is covered.  

I'm not sure whether part of what you are getting at is actually speaking to success criteria or if there is another user need.  Are you saying that the user need is not just that users need each of these individual items but that they need all of them together?

**Comment by allanj-uaag** [comment]
Though wordy, the thing that pops out to me is 
Users must be capable of adjusting the visual interface to meet their exact visual needs.
this seems like a principle.  The "User can..." user needs define and/or expands the principle at a granular level.
The rest is explanation and recapitulates most if not all of the "User can ..." user needs. 
Wayne, is there anything in your information that is not covered in User Needs document?

**Comment by shawna-slh** [comment]
Thanks, Wayne, for the organized perspective
 This sounds a lot like [GPII](http://gpii.net/About.html).

I think what we need to figure out is:
- Short-term: How to we address user needs in the upcoming W3C WAI work on WCAG?
- Longer-term: How can the next generation of W3C accessibility standards (WAI.Next) better address the issues across user agents and content?

**Comment by lauracarlson** [comment]
I agree Shawn. Short term we need to figure out some SCs and techniques that cover whatever gaps we can cover. Long term the answer may be GPII.

BTW a related paper (2005)  [Personalization of Web browsing: adaptations to meet the needs of older adults](http://download.springer.com/static/pdf/108/art%253A10.1007%252Fs10209-005-0110-9.pdf?originUrl=http%3A%2F%2Flink.springer.com%2Farticle%2F10.1007%2Fs10209-005-0110-9&token2=exp=1465483819~acl=%2Fstatic%2Fpdf%2F108%2Fart%25253A10.1007%25252Fs10209-005-0110-9.pdf%3ForiginUrl%3Dhttp%253A%252F%252Flink.springer.com%252Farticle%252F10.1007%252Fs10209-005-0110-9*~hmac=7c2ceb34414df4484968d942e4766365a6a24e9562d3ff302180723140e2c7b3)  discusses a software application that allows users to control their Web environment.
